File: HesiasFleet/HesiasFleet.Mobile/app/src/main/java/fr/hesias/fleet/data/model/Note.kt
package fr.hesias.fleet.data.model

/** NoteDto renvoyé par l'API */
data class Note(
    val id: Int,
    val vehicleId: Int,
    val content: String,
    val date: String,
    val mileage: Int? = null,
    val operationId: Int? = null,
    val metaOperationId: Int? = null
)

File: HesiasFleet/HesiasFleet.Core/Entities/Note.cs
namespace HesiasFleet.Core.Entities;

public class Note
{
    public int Id { get; set; }

    public string Content { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public int? Mileage { get; set; }

    // note always belongs to a vehicle (an entry in its log)
    public int VehicleId { get; set; }
    public Vehicle Vehicle { get; set; } = null!;

    // optionally linked to an operation or a meta-operation
    public int? OperationId { get; set; }
    public Operation? Operation { get; set; }

    public int? MetaOperationId { get; set; }
    public MetaOperation? MetaOperation { get; set; }
}